storage/storage_integration_tests/storage_http_tests.cpp

```diff
@@ -0,0 +1,110 @@
+#include "testing/testing.hpp"
+
+#include "storage/storage.hpp"
+
+#include "platform/platform.hpp"
+
+#include "platform/platform_tests_support/scoped_dir.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "base/scope_guard.hpp"
+#include "base/string_utils.hpp"
+#include "base/thread.hpp"
+
+#include "std/string.hpp"
+
+#include <QtCore/QCoreApplication>
+
+using namespace platform;
+using namespace storage;
+
+class WritableDirChanger
+{
+  static string const kMapTestDir;
+  string const m_writableDirBeforeTest;
+  string const m_mapTestDirFullPath;
+
+public:
+  WritableDirChanger()
+    : m_writableDirBeforeTest(GetPlatform().WritableDir())
+    , m_mapTestDirFullPath(m_writableDirBeforeTest + kMapTestDir)
+  {
+    Platform & platform = GetPlatform();
+    platform.MkDir(m_mapTestDirFullPath);
+    platform.SetWritableDirForTests(m_mapTestDirFullPath);
+  }
+  ~WritableDirChanger()
+  {
+    Platform & platform = GetPlatform();
+    string const writableDirForTest = platform.WritableDir();
+    platform.SetWritableDirForTests(m_writableDirBeforeTest);
+    platform.RmDir(writableDirForTest);
+  }
+};
+string const WritableDirChanger::kMapTestDir("map-tests");
+
+void ChangeCountryFunctionForAngola(TCountryId const & countryId) {}
+
+void ProgressFunctionForAngola(TCountryId const & countryId, LocalAndRemoteSizeT const & mapSize)
+{
+  if (mapSize.first != mapSize.second)
+    return;
+
+  QCoreApplication::exit();
+}
+
+void UpdateForAngola(platform::LocalCountryFile const & localCountryFile) {}
+
+UNIT_TEST(StorageDownloadNodeAndDeleteNodeTests)
+{
+  Storage storage;
+  if (!version::IsSingleMwm(storage.GetCurrentDataVersion()))
+    return;  // Test is valid for single mwm case only.
+
+  WritableDirChanger writableDirChanger;
+  UNUSED_VALUE(writableDirChanger);
+
+  storage.Init(UpdateForAngola);
+  storage.RegisterAllLocalMaps();
+  storage.Subscribe(ChangeCountryFunctionForAngola, ProgressFunctionForAngola);
+  storage.SetDownloadingUrlsForTesting({"http://eu1.mapswithme.com/"});
+  string const angolaCountryId = "Angola";
+  string const versionStr = strings::to_string(storage.GetCurrentDataVersion());
+  platform::tests_support::ScopedDir cleanupVersionDir(versionStr);
+  UNUSED_VALUE(cleanupVersionDir);
+  MY_SCOPE_GUARD(cleanupAngola,
+                 bind(&Storage::DeleteCountry, &storage, angolaCountryId, MapOptions::Map));
+
+  string const angolaMwmFullPath = my::JoinFoldersToPath({GetPlatform().WritableDir(), versionStr},
+                                                         angolaCountryId + DATA_FILE_EXTENSION);
+  string const angolaDownloadingFullPath = my::JoinFoldersToPath(
```

Comment:
Hm, sorry, I still do not understand your answer. I propose you to call DeleteDownloaderFilesForCountry from platform/local_country_file_utils.hpp instead of an explicit check that downloader files do not exists in a test directory. All your angolaResumeFullPath and angolaDownloadingFullPath variables will gone away. Moreover, it's not correct to expect that downloader files do not exists in a test directory, because it's possible that the test directory exists with some garbage at the time the unit test is started (due to crashes during previous runs of tests).